client/pool.go

```diff
@@ -151,137 +152,170 @@ func (p *ClientPool) listenChans(ctx context.Context) {
 
 // executeRequest handles making the HTTP request with server rotation and retires. Only if all servers return an
 // error will this method return an error.
-func (p *ClientPool) executeRequest(client *blockbook.BlockBookClient, queryFunc func(c *blockbook.BlockBookClient) error) error {
+func (p *ClientPool) executeRequest(queryFunc func(c *blockbook.BlockBookClient) error) error {
 	for e := p.newMaximumTryEnumerator(); e.next(); {
+		var client = p.poolManager.AcquireCurrent()
```

Comment:
Should this be `AcquireCurrentWhenReady`?

Reply:
Yes, thanks.